pkg/x/http/transport.go

```diff
@@ -68,10 +70,11 @@ func NewTransport(opts Options) http.RoundTripper {
 	}
 	tr.DialContext = d.DialContext
 
-	// Configure TLS
-	if opts.Insecure {
+	// Configure TLS only when needed.
+	if opts.CaCerts != nil || opts.Insecure {
 		tr.TLSClientConfig = &tls.Config{
 			InsecureSkipVerify: opts.Insecure,
+			RootCAs:            opts.CaCerts,
```

Comment:
Is it fine to pass nil here? Does `net/http` transparently populate the default root CAs?